statusdb_snicuser_checker.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python
+"""
+Get open projects from statusdb, checks if the users have a SNIC account and
+writes the result back into statusdb.
+"""
+import argparse
+import yaml
+from couchdb import Server
+import requests
+from requests.auth import HTTPBasicAuth
+import json
+
+def update_statusdb(config, dryrun=True):
+    url_string = 'http://{}:{}@{}:{}'.format(config['statusdb'].get('username'), config['statusdb'].get('password'),
+                                              config['statusdb'].get('url'), config['statusdb'].get('port'))
+    couch = Server(url=url_string)
+    assert couch, 'Could not connect to {}'.format(settings.get('url'))
+
+    proj_db = couch['projects']
+    open_projs = proj_db.view('project/summary',include_docs=True, descending=True)[['open','Z']:['open','']]
+
+    for project in open_projs:
+        doc = project.doc
+        flag = False
+        if project.value.get('delivery_type') == 'GRUS':
+            if project.value['details'].get('snic_checked'):
+                if not project.value['details']['snic_checked']['status']:
+                    email = project.value['order_details']['fields'].get('project_pi_email')
+                    check = snic_check(email, config['SNIC'])
+                    if check:
+                        doc['details']['snic_checked']['status'] = check
+                        flag = True
+
+            else:
+                snic_checked = {'status': True}
+                #roles = ['project_lab_email','project_bx_email', 'project_pi_email']
+                if project.value.get('order_details'):
+                    email = project.value['order_details']['fields'].get('project_pi_email')
+                    if email:
+                        if not snic_check(email, config['SNIC']):
+                            snic_checked['status'] = False
+                    #Add the new field to project details
+                    doc['details']['snic_checked'] = snic_checked
+                    flag = True
+        #write to projects doc
+        if flag:
+            if not dryrun:
+                proj_db.save(doc)
+            else:
+                print(doc['project_name'], doc['details']['snic_checked'])
+
+def snic_check(email, config):
+    url = 'https://supr.snic.se/api/person/email_present/?email={}'.format(email)
+    response = requests.get(url, auth=HTTPBasicAuth(config.get('username'), config.get('password')))
+    return json.loads(response.content)['email_present']
```

Comment:
Could you stick in a check here that will tell the user if the SNIC API doesn't allow access. As it was IP restricted I got an error.

Reply:
Done!